src/compiler/watchUtilities.ts

```diff
@@ -34,9 +34,20 @@ namespace ts {
         clearCache(): void;
     }
 
+    type Canonicalized = string & { __canonicalized: void };
+
     interface MutableFileSystemEntries {
         readonly files: string[];
         readonly directories: string[];
+        sortedAndCanonicalizedFiles?: undefined | Canonicalized[]

```

Comment:
Fun fact: files and directories are already sorted (just not canonically) does it make sense to direct look and create sorted list only if we couldnt look.. in theory most of the times filenames that we are looking up should match and hence might not need the sorted canonical list ?

Reply:
Sorry, I'm not sure I understand your suggestion.  Maybe you're saying we should search for the un-canonicalized name in the unsorted (by us) list before we do anything more expensive?

Reply:
Also, won't that semi-predictable order fall apart when a new entry is added (unless we switch to something fancier than `push`)?